### ISSUE TITLE
Simplify setting up the NLog config 

### ISSUE THREAD

**Issue by 304NotModified** [NLog/NLog.Web#184]
Simplify setting up NLog for ASP.NET Core 2

Currently setting up takes a lot of steps. It should be as easy as possible  - but also flexible enough. Also the current design not really suitable for ASP.NET Core 2

Current config (code set-up only):

```c#
using NLog.Extensions.Logging;
using NLog.Web;

public Startup(IHostingEnvironment env)
{
    env.ConfigureNLog("nlog.config");
}

public void ConfigureServices(IServiceCollection Services)
{
    //call this in case you need aspnet-user-authtype/aspnet-user-identity
    services.AddSingleton<IHttpContextAccessor, HttpContextAccessor>();
}

// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
{

    //add NLog to ASP.NET Core
    loggerFactory.AddNLog();

    //add NLog.Web
    app.AddNLogWeb();

   //note: remove the old loggerFactory, like loggerFactory.AddConsole and  loggerFactory.AddDebug
```

Redesign the API design for ASP.NET Core 2

(maybe also replace `ConfigureNLog(this IHostingEnvironment env`)

**Comment by 304NotModified** [comment]
@grokky1 

I think it will be something like this, do you think this is simple enough?

```c#
  public class Program
    {
        public static void Main(string[] args)
        {
            // NLog: setup the logger first to catch all errors
            LogManager.Configuration = new XmlLoggingConfiguration("NLog.config");
            var logger = LogManager.GetCurrentClassLogger();
            try
            {
                BuildWebHost(args).Run();
            }
            catch (Exception e)
            {
                //NLog: catch setup errors
                logger.Error(e, "Stopped program because of exception");
                throw;
            }
        }

        public static IWebHost BuildWebHost(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .UseNLog() // NLog: setup NLog for Dependency injection
                .Build();
    }

```

**Comment by 304NotModified** [comment]
i'm doubting to create a helper for 

```C#
LogManager.Configuration = new XmlLoggingConfiguration("NLog.config");
var logger = LogManager.GetCurrentClassLogger();
```

But maybe it's needed (for #187 at least)

**Comment by grokky1** [comment]
Core2 is much simpler than before, but honestly, some stuff is still confusing, and there's too many places to tweak to get stuff done.

So, I like your convenience functions (`UseNLog` and `InitLogger`), because at least logging will be easy to setup! 👍  Also `UseNLog()` is named similarly to other stuff in the webhost builder so that's good. (@xperiandri : I think we discussed this a little while ago as well?)

As far as your comment `//note: this also adds the default MS loggers` - the default builder adds MS loggers, even in production (debug and console), but these [can be removed](https://stackoverflow.com/a/46336988/7059613). Maybe a note in the docs for users who might want to do that, because most users won't even realise this.

**Comment by 304NotModified** [comment]
@grokky1 thanks for the info

> but these can be removed. 

Thanks! Needed that info :)

If have now (locally) `var logger = NLogStart.InitLogger("NLog.config");`, but not sure about the name of the class (`NLogStart`). What do you think?

**Comment by grokky1** [comment]
Pity we can't use `NLog` because that's the name of the namespace.

But I suppose it's fine. In truth we will only use it once in the app's lifecycle.

**Comment by 304NotModified** [comment]
fixed by  #191
